### PR TITLE
feat(elec): First A380 electrical system implementation

### PR DESCRIPTION
**Pull request by Gurgel100**
Fixes #[issue_no]

## Summary of Changes

This PR adds the first implementation of the electrical system of the A380.

## Additional context

Discord username (if different from GitHub):

## Testing instructions

Can only be tested internally.

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

File: fbw-a380x/src/wasm/systems/a380_systems/src/electrical/alternating_current.rs [hunk 1]
```diff
@@ -1,119 +1,133 @@
...
+    // This is actually a 2-way switch but we simulate this with 2 contactors
+    ac_emer_contactor: [Contactor; 2],
+    eha_contactors: [Contactor; 2],
+    tr_apu: TransformerRectifier,
     emergency_gen_contactor: Contactor,
-    static_inv_to_ac_ess_bus_contactor: Contactor,
-    ac_stat_inv_bus: ElectricalBus,
-    ac_gnd_flt_service_bus: ElectricalBus,
-    ext_pwr_to_ac_gnd_flt_service_bus_and_tr_2_contactor: Contactor,
+    //ac_gnd_flt_service_bus: ElectricalBus,
```
Comment (crocket63) [inline fbw-a380x/src/wasm/systems/a380_systems/src/electrical/alternating_current.rs:32]: Still need this commented line?

File: fbw-a380x/src/wasm/systems/a380_systems/src/electrical/alternating_current.rs [hunk 2]
```diff
@@ -1,119 +1,133 @@
...
                 context,
-                ElectricalBusType::AlternatingCurrentGndFltService,
+                ElectricalBusType::AlternatingCurrentNamed("247XP"),
             ),
-            ext_pwr_to_ac_gnd_flt_service_bus_and_tr_2_contactor: Contactor::new(context, "12XN"),
+            ac_emer_contactor: [1, 2].map(|i| Contactor::new(context, &format!("3XB.{i}"))),
+            eha_contactors: ["911XN", "911XH"].map(|id| Contactor::new(context, id)),
+            tr_apu: TransformerRectifier::new(context, 4),
+            emergency_gen_contactor: Contactor::new(context, "5XE"),
+            // ac_gnd_flt_service_bus: ElectricalBus::new(
```
Comment (crocket63) [inline fbw-a380x/src/wasm/systems/a380_systems/src/electrical/alternating_current.rs:59]: Same remark :)

File: fbw-a380x/src/wasm/systems/a380_systems/src/electrical/alternating_current.rs [hunk 3]
```diff
@@ -129,145 +143,60 @@ impl A380AlternatingCurrentElectrical {
...
-
-        self.ac_ess_feed_contactors.update(
-            context,
-            electricity,
-            &self.ac_bus_1,
-            &self.ac_bus_2,
-            overhead,
-        );
-
-        self.ac_ess_feed_contactors
```
Comment (crocket63) [inline fbw-a380x/src/wasm/systems/a380_systems/src/electrical/alternating_current.rs:152]: Is this planned for later?

**Review by crocket63 — COMMENTED** [review]
Great!

File: fbw-a380x/src/wasm/systems/a380_systems/src/electrical/alternating_current.rs (hunk 3 above)
Comment (Gurgel100) [reply fbw-a380x/src/wasm/systems/a380_systems/src/electrical/alternating_current.rs:152]: Nope. Implemented now

File: .github/CHANGELOG.md [hunk 4]
```diff
@@ -11,6 +11,7 @@
 1. [EFB] Added pause at T/D function - @2hwk (2Cas#1022)
 1. [FMS] Add a default RNP of 0.3 for RNP AR procedure legs - @tracernz (Mike)
 1. [SD] Corrected fuel transfer valve amber logic - @tracernz (Mike)
+1. [ELEC] Added electrical system for A380 - @Gurgel100 (Pascal)
```
Comment (crocket63) [inline .github/CHANGELOG.md:14]: So I think for now we do not put 380 changes  in there

Comment (Gurgel100) [reply .github/CHANGELOG.md]: Removed the changelog entry

**Review by crocket63 — APPROVED** [review]
Amazing piece of work :)

**Pull request merged**